libs/sdk-ui-dashboard/src/model/commandHandlers/layout/moveLayoutSectionHandler.ts

```diff
@@ -3,8 +3,51 @@ import { SagaIterator } from "redux-saga";
 import { DashboardContext } from "../../types/commonTypes";
 import { MoveLayoutSection } from "../../commands";
 import { dispatchDashboardEvent } from "../../eventEmitter/eventDispatcher";
-import { commandRejected } from "../../events/general";
+import { invalidArgumentsProvided } from "../../events/general";
+import { selectLayout } from "../../state/layout/layoutSelectors";
+import { put, select } from "redux-saga/effects";
+import { validateSectionExists, validateSectionPlacement } from "./validation/layoutValidation";
+import { layoutActions } from "../../state/layout";
+import { layoutSectionMoved } from "../../events/layout";
 
 export function* moveLayoutSectionHandler(ctx: DashboardContext, cmd: MoveLayoutSection): SagaIterator<void> {
-    yield dispatchDashboardEvent(commandRejected(ctx, cmd.correlationId));
+    const layout: ReturnType<typeof selectLayout> = yield select(selectLayout);
+    const { sectionIndex, toIndex } = cmd.payload;
+
+    if (!validateSectionExists(layout, sectionIndex)) {
+        return yield dispatchDashboardEvent(
+            invalidArgumentsProvided(
+                ctx,
+                `Attempting to move non-existent section from index ${sectionIndex}. There are only ${layout.sections.length} sections.`,
+                cmd.correlationId,
+            ),
+        );
+    }
+
+    if (!validateSectionPlacement(layout, toIndex)) {
+        return yield dispatchDashboardEvent(
+            invalidArgumentsProvided(
+                ctx,
+                `Attempting to move section to a wrong index ${toIndex}. There are currently ${layout.sections.length} sections.`,
+                cmd.correlationId,
+            ),
+        );
+    }
+
+    yield put(
+        layoutActions.moveSection({
+            sectionIndex,
+            toIndex,
```

Comment:
Should we use `absoluteIndex` here and make the action accept only non-negative indices?